### PR TITLE
Fix flaky rest-basics test

### PR DESCRIPTION
**Pull request by Janpot**
Part of the rest-basics test is still trying to access httpbin.org instead of our own hosted instance. This leads to flaky tests as in https://app.circleci.com/pipelines/github/mui/mui-toolpad/5219/workflows/f98e7c8d-ff94-48c5-a47b-488ae4c8090f/jobs/21186

In this PR I'm making sure that
- All httpbin urls point to our own instance
- Both playwright and toolpad container can access the httpbin service under the same host

### CODE REVIEW

File: packages/toolpad-app/src/server/appTemplateDoms/doms.ts [hunk 1]
```diff
@@ -17,9 +17,8 @@ const appTemplates: Record<AppTemplateId, () => Promise<appDom.AppDom | null>> =
     getAppTemplateDomFromPath(config.isDemo ? './images-demo.json' : './images.json'),
   default: async () => {
     const template = await getAppTemplateDomFromPath('./default.json');
-    (
-      template.nodes['1313wn3' as NodeId] as appDom.QueryNode
-    ).attributes.query.value.url.value = `${config.externalUrl}/static/employees.json`;
```
Comment (Janpot) [inline packages/toolpad-app/src/server/appTemplateDoms/doms.ts:22]: Never use string manipulation to resolve urls

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -37,7 +37,7 @@ If you would like to hack on MUI Toolpad or want to run the latest version, you
    TOOLPAD_DATABASE_URL=postgres://postgres:postgres@localhost:5432/postgres
    # For a custom port:
    # PORT=3004
-   # EXTERNAL_URL=http://localhost:3004/
```
Comment (Janpot) [inline CONTRIBUTING.md:40]: `EXTERNAL_URL` doesn't exist

File: docker-compose.test.yml [hunk 3]
```diff
@@ -1,4 +1,4 @@
-version: '3.7'
+version: '3.9'
```
Comment (Janpot) [inline docker-compose.test.yml:1]: So we can use `profiles` which prevents the `playwright` container from starting on `up`

File: docker-compose.test.yml [hunk 4]
```diff
@@ -7,10 +7,11 @@ services:
       - TOOLPAD_DATABASE_URL=postgresql://postgres:postgres@postgres:5432/postgres?connect_timeout=10
       - TOOLPAD_ENABLE_CREATE_BY_DOM=1
       - PORT=3000
-      - EXTERNAL_URL=http://localhost:3000/
-      - TOOLPAD_EXTERNAL_URL=http://172.17.0.1:3000/
+      - TOOLPAD_EXTERNAL_URL=http://toolpad:3000/
```
Comment (Janpot) [inline docker-compose.test.yml:10]: We'll run the playwright container from the same compose file, so both the toolpad container and playwright container are in the same network. So both on server and client `http://toolpad:3000/` will resolve

File: test/integration/rest-basic/index.spec.ts [hunk 5]
```diff
@@ -25,17 +26,17 @@ test.use({
 });
 
 test('rest basics', async ({ page, browserName, api }) => {
-  const dom = await readJsonFile(path.resolve(__dirname, './restDom.json'));
+  const dom = await readJsonFile(path.resolve(__dirname, './restDom.json'), (key, value) => {
+    if (typeof value === 'string') {
+      return value.replaceAll(HTTPBIN_SOURCE_URL, HTTPBIN_TARGET_URL);
```
Comment (Janpot) [inline test/integration/rest-basic/index.spec.ts:31]: Use a reviver to replace all strings containing the `HTTPBIN_SOURCE_URL`

File: test/utils/fs.ts [hunk 6]
```diff
@@ -1,6 +1,8 @@
 import * as fs from 'fs/promises';
 
-export async function readJsonFile(path: string): Promise<any> {
+export type Reviver = NonNullable<Parameters<typeof JSON.parse>[1]>;
+
+export async function readJsonFile(path: string, reviver?: Reviver): Promise<any> {
   const content = await fs.readFile(path, { encoding: 'utf-8' });
-  return JSON.parse(content);
+  return JSON.parse(content, reviver);
```
Comment (bharatkashyap) [inline test/utils/fs.ts:7]: TIL: There is a second argument to JSON.parse :)

**Review by bharatkashyap — APPROVED** [review]

**Pull request merged**